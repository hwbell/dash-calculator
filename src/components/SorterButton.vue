<template>
  <div
    class="sorter-button-component"
  >
      <button @click="sort(sorter)" class="btn btn-sm btn-outline-primary">
        {{ text }}
      </button>
  </div>
</template>

<script>
export default {
  name: "SorterButton",
  props: {
    text: String,
    sorter: String,
    sort: Function
  },
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
.scroll-button-component {
  margin: 4px;
}
.btn-sm {
  margin: 4px;
  font-weight: bolder;
  /* width: 140px; */
}
</style>
