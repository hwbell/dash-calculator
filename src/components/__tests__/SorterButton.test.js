import { mount } from '@vue/test-utils';
import SorterButton from '../SorterButton.vue';

const moment = require('moment');
const lastDay = moment()
  .subtract(1, "d")
  .format("YYYY-MM-DD");

const sortStub = jest.fn();

describe('SorterButton', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = mount(SorterButton, {
      propsData: {
        text: 'last day',
        sorter: lastDay,
        sort: sortStub
      }
    })
  })

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  test('renders a button for each object in the array map', () => {
    expect(wrapper.findAll('.sorter-button-component').length).toBe(1);
    expect(wrapper.findAll('.btn-outline-primary').length).toBe(1);
  })

  test('receives and uses the props', () => {
    expect(wrapper.vm.text).toBe('last day');

    wrapper.vm.sort();
    expect(sortStub).toHaveBeenCalledTimes(1);

  })
})