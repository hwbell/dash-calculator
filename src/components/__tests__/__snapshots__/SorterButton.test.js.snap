// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SorterButton renders correctly 1`] = `
<div
  class="sorter-button-component"
>
  <button
    class="btn btn-sm btn-outline-primary"
  >
    
      last day
    
  </button>
</div>
`;
